src/dirichlet_multi.jl

```diff
@@ -38,3 +39,6 @@ function posterior_canon(pri::Dirichlet, G::Type{Multinomial}, x::Matrix{T}, w::
 	DirichletCanon(a)
 end
 
+# missing method
+posterior(pri::Dirichlet, G::IncompleteFormulation, x) = convert(typeof(pri), posterior_canon(pri, G, x))
```

Comment:
The trouble is here, this doesn't see 

https://github.com/JuliaStats/Distributions.jl/blob/c431d20d3a0f00e0cf22f9fd959b9bb11d978517/src/multivariate/dirichlet.jl#L56

```julia
convert(::Type{Dirichlet{T}}, cf::DirichletCanon) where {T<:Real} =
    Dirichlet(convert(AbstractVector{T}, cf.alpha))
```


```
MethodError: Cannot `convert` an object of type Distributions.DirichletCanon{Float64,Array{Float64,1}} 
to an object of type Dirichlet{Float64,Array{Float64,1},Float64}
```

Reply:
`typeof(pri)` contains additional type parameters AFAICT. That is, this could be fixed by defining `convert(::Type{<:Dirichlet{T}}, ...)` in Distributions. I'm not sure if that's a good idea though, it feels like if you specify the additional type parameters you should also get something with these requested type parameters, otherwise it might break the (implicit) assumptions of `convert`. So I'd suggest to fix it by changing this line to
```suggestion
function posterior(pri::Dirichlet{T}, G::IncompleteFormulation, x) where {T<:Real}
    return convert(Dirichlet{T}, posterior_canon(pri, G, x))
end
```